### PR TITLE
Qca dev

### PR DESCRIPTION
**Pull request by agary-ucsd**
Closed previous pull request because refactoring caused an invalid import.  This pull request resolves this issue

### CODE REVIEW

File: integration/test_dtda.in [hunk 1]
```diff
@@ -8,8 +8,18 @@
 (FAILURE :reason DRUG_NOT_FOUND)
 (FIND-DISEASE-TARGETS :disease NIL)
 (FAILURE :reason INVALID_DISEASE)
+
```
Comment (bgyori) [inline integration/test_dtda.in]: It looks like these extra lines were added by accident - if that's the case could you revert the changes to this file?

File: bioagents/qca/qca_module.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        reply_content = KQMLList.from_string(
+            '(SUCCESS :paths (' + path_statements + '))')
+
+        return reply_content
+
+        #reply_msg = KQMLPerformative('reply')
+        #reply_msg.set_parameter(':content', reply_content)
+        #self.reply(msg, reply_msg)
+
+    def respond_find_qca_path_template(self, content_list):
```
Comment (bgyori) [inline bioagents/qca/qca_module.py]: This seems like something copied from the DTDA code, can it be removed or is it needed here?

File: integration/test_qca.in [hunk 3]
```diff
@@ -0,0 +1,6 @@
+(FIND-QCA-PATH : TARGET ("SHC1") : SOURCE ("IRS1") : RELTYPE ("controls-state-change-of"))
```
Comment (bgyori) [inline integration/test_qca.in]: In the message format I think the name of an argument has to follow the : directly without a space in between, for instance
```(FIND-QCA-PATH :TARGET ("SHC1") :SOURCE ("IRS1") :RELTYPE ("controls-state-change-of"))```

File: bioagents/qca/qca.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+_resource_dir = os.path.dirname(os.path.realpath(__file__)) + '/../resources/'
+
+class PathNotFoundException(Exception):
+    def __init__(self, *args, **kwargs):
+            Exception.__init__(self, *args, **kwargs)
+
+class QCA:
+    def __init__(self):
+        logger.debug('Starting QCA')
+        self.host = "http://www.ndexbio.org"
```
Comment (bgyori) [inline bioagents/qca/qca.py:24]: You could consider passing these (host names, credentials, etc) as arguments on startup that can be integrated into the central system startup script. This way these potentially sensitive details won't need to be hard coded and published.

File: integration/test_qca.in (hunk 3 above)
Comment (bgyori) [reply integration/test_qca.in]: A few other questions about the message format: how does the QCA respond if no path is found? Presumably ```(SUCCESS :paths ())```? Is the response the same of either of the nodes is not found in the network(s) at all?

File: integration/test_qca.in [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+(SUCCESS :paths (((IRS1 (:93584034 (:directed True :interaction controls-state-change-of)) HRAS (:93581981 (:directed True :interaction controls-state-change-of)) RHOA (:93594410 (:directed True :interaction controls-state-change-of)) SHC1) (IRS1 (:93584046 (:directed True :interaction controls-state-change-of)) PIK3CA (:93590756 (:directed True :interaction controls-state-change-of)) RHOA (:93594410 (:directed True :interaction controls-state-change-of)) SHC1))))
```
Comment (bgyori) [inline integration/test_qca.in]: Is :93584034 an ID for the edge here? I think it would be better to use : for a keyword whose argument could be 93584034, for instance (:id 93584034 (:directed True ...))

**Comment by agary-ucsd** [comment]
Yes, this is a good point.  I will pull the credentials out.

Thanks,
Aaron Gary
Bioinformatics Software Engineer
Center for Computational Biology and Bioinformatics
School of Medicine
University of California San Diego

**Comment by agary-ucsd** [comment]
Hey Ben,
I decided to remove the edge id altogether.  Now it will return an array of edges instead of an id-keyed dictionary of edges.  I can’t see any purpose in including the edge id since it’s an internal identifier.  That’s at least my assumption at this point.  Here is an example of the new return schema:

(CALM1 ((:directed True :pmid (1377362 1715516) :interaction controls-transport-of) (:directed True :pmid (1377362 1715516) :interaction controls-phosphorylation-of)) NFATC2)

Thanks,
Aaron Gary
Bioinformatics Software Engineer
Center for Computational Biology and Bioinformatics
School of Medicine
University of California San Diego

File: integration/test_qca.in [hunk 6]
```diff
@@ -0,0 +1,6 @@
+(FIND-QCA-PATH :TARGET ("SHC1") :SOURCE ("IRS1") :RELTYPE ("controls-state-change-of"))
+(SUCCESS :paths (((IRS1 ((:directed True :interaction controls-state-change-of)) HRAS ((:directed True :interaction controls-state-change-of)) RHOA ((:directed True :interaction controls-state-change-of)) SHC1) (IRS1 ((:directed True :interaction controls-state-change-of)) PIK3CA ((:directed True :interaction controls-state-change-of)) RHOA ((:directed True :interaction controls-state-change-of)) SHC1))))
+(HAS-QCA-PATH :TARGET ("SHC1") :SOURCE ("IRS1"))
+(SUCCESS :haspath (True))
+(FIND-QCA-PATH :TARGET ("CALM1") :SOURCE ("NFATC2"))
+(SUCCESS :paths (((IRS1 (:108 (:interaction in-complex-with)) SHC1) (IRS1 (:93584068 (:interaction in-complex-with)) SHC1))))
```
Comment (bgyori) [inline integration/test_qca.in]: I don't think this last test makes sense here - shouldn't it be a path between NFATC2 and CALM1?
I got this when running the system: (SUCCESS :paths (((NFATC2 ((:directed True :interaction controls-expression-of)) IFNG ((:directed True :interaction controls-state-change-of)) CALM1) (NFATC2 ((:directed True :interaction controls-expression-of)) CXCL8 ((:directed True :interaction controls-state-change-of)) GNA14 ((:directed True :interaction controls-state-change-of)) CALM1))))

Comment (agary-ucsd) [reply integration/test_qca.in]: Yes, I need to update the test.  I replaced the input with a different use case, but failed to update the expected output.
